### ISSUE TITLE
npm run build: Cannot find module '/home/username/destreamer/node_modules/typescript/bin/tsc

### ISSUE THREAD

**Issue by happybear88** [snobu/destreamer#90]
**Env**
Ubuntu 16.04 on WSL Windows 10 1909
git version 2.7.4
node v14.0.0
npm 6.14.4

**Steps**
```
git clone https://github.com/snobu/destreamer
cd destreamer
npm install
npm run build
```
ER: success
AR: error
```
> destreamer@2.0.0 build /home/username/destreamer
> echo Transpiling TypeScript to JavaScript... & node node_modules/typescript/bin/tsc --listEmittedFiles

Transpiling TypeScript to JavaScript...
internal/modules/cjs/loader.js:1017
  throw err;
  ^

Error: Cannot find module '/home/username/destreamer/node_modules/typescript/bin/tsc'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1014:15)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! destreamer@2.0.0 build: `echo Transpiling TypeScript to JavaScript... & node node_modules/typescript/bin/tsc --listEmittedFiles`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the destreamer@2.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-24T07_10_52_463Z-debug.log
```

Log attached.
[2020-04-24T07_10_52_463Z-debug.log](https://github.com/snobu/destreamer/files/4527284/2020-04-24T07_10_52_463Z-debug.log)

**Comment by snobu** [comment]
It won't work in WSL, puppeteer can't pop up a browser window. Run it with Windows Node.js in cmd.exe or PowerShell.

**Comment by happybear88** [comment]
Did I miss this in readme? If not, please consider documenting it.

**Comment by snobu** [comment]
You didn't, we left it out by accident. I'll add a note for it right now. Thanks for reporting this one.

**Comment by snobu** [comment]
Quick update here, i've tested this on WSL2 and it does work, however you'll need an Xserver so puppeteer can launch the Linux-flavored Chromium that comes down with npm install.

```sh
# For WSL1
# Start Xming on Windows
export DISPLAY=:0
./destreamer.sh -i URL_TO_VIDEO

# For WSL2
# Start Xming on Windows
export DISPLAY=$(grep nameserver /etc/resolv.conf | cut -f2 -d ' '):0
./destreamer.sh -i URL_TO_VIDEO
```
Xming -
https://sourceforge.net/projects/xming/

**Comment by MauroFVieira** [comment]
Just confirming that @snobu 's comment about using your Windows version of Chrome (below) does work.
```
const browser = await puppeteer.launch({
        executablePath: "/mnt/c/path/to/your/chrome.exe"
        headless: false,
        args: ['--disable-dev-shm-usage']
    });
```

The XServer solution was not working for me, as the Chromium window was not rendering any content.

The only caveat is that you need to manually close the Chrome window after a successful login, as the script will fail to do it.

**Comment by hawnzah** [comment]
Hi, I'm trying to run destreamer on wsl2.
I changed the executablePath to my windows chrome.exe, but it is not rendering any content.

In console I get this error:

[WARNING] .token_cache not found.
Launching headless Chrome to perform the OpenID Connect dance...
[ERROR] [object Object]

Do you know what could be the problem?

**Comment by rockguy40** [comment]
> Hi, I'm trying to run destreamer on wsl2.
> I changed the executablePath to my windows chrome.exe, but it is not rendering any content.
> 
> In console I get this error:
> 
> [WARNING] .token_cache not found.
> Launching headless Chrome to perform the OpenID Connect dance...
> [ERROR] [object Object]
> 
> Do you know what could be the problem?

I have the same problem. I can't seem to make it work even with xming

**Comment by phormula** [comment]
> 
> 
> Just confirming that @snobu 's comment about using your Windows version of Chrome (below) does work.
> 
> ```
> const browser = await puppeteer.launch({
>         executablePath: "/mnt/c/path/to/your/chrome.exe"
>         headless: false,
>         args: ['--disable-dev-shm-usage']
>     });
> ```
> 
> The XServer solution was not working for me, as the Chromium window was not rendering any content.
> 
> The only caveat is that you need to manually close the Chrome window after a successful login, as the script will fail to do it.

I think you need to make a double slash 
Like this 
```const browser = await puppeteer.launch({
        executablePath: "//mnt//c//path//to//your//chrome.exe"
        headless: false,
        args: ['--disable-dev-shm-usage']
    });
```

**Comment by rockguy40** [comment]
I ended up using this tutorial on yt, works as a charm [https://www.youtube.com/watch?v=9Fx7UCsSG9g](url)

**Comment by bogdan-calapod** [comment]
I managed to get it working with WSL2 and GWSL (from the Microsoft Store). I had to install these dependencies beforehand:

```
sudo apt install -y libatk-bridge2.0-dev libatk1.0-dev libnss3-dev libcups2-dev libgtk-3-dev
```

I'm using an Ubuntu 20.04 distro from the store. After installing GWSL so that the X server is configured, as well as the dependencies above that are required to run GTK apps, it's all working smoothly. 

I can make a PR that runs these steps / suggests GWSL if WSL2 environments are detected.

Thanks for the tool, guys! It's really useful!
